### ISSUE TITLE
Follow-up-status doesn't change after the end of the follow-up-period

### ISSUE THREAD

**Issue by Marko-ilmkreis** [hzi-braunschweig/SORMAS-Project#3791]
### Bug Description

I don't know if this is really a bug or can be solved by different way.

In case- and contact directory if shows that the persons are under follow-up, even when the follow-up period already ends:

![grafik](https://user-images.githubusercontent.com/74791118/102211071-59c47180-3ed3-11eb-8c50-8264b41761aa.png)

There is also no option to set the follow up to be finished. We have this with imported cases and new entered cases.

### Steps to Reproduce
1. open the case and/or contact directory

### Expected Behavior

If the last date of the follow-up-period is finished, the follow-up must be finished in the system too.

**Comment by MateStrysewske** [comment]
The follow-up status is only changed to "Follow-up completed" if there actually was a visit on the last day of follow-up. Otherwise, the follow-up is extended because the follow-up process was not finished appropriately.

**Comment by Marko-ilmkreis** [comment]
@MateStrysewske do you mean with visit that someone take a call on the last day and registered it in Sormas? As the phone calls are not yet organized through Sormas we would never end the follow up.

**Comment by MateStrysewske** [comment]
Exactly. If that's not the case, we probably need a server configuration that disables this logic. I'll turn this issue into a feature request and reopen it.

**Comment by Marko-ilmkreis** [comment]
@MateStrysewske Thanks! Do you have any idea how to set all oustanding follow-ups on finished at the moment?

**Comment by MateStrysewske** [comment]
There's no way to bulk-set the follow-up status to completed unfortunately; it's possible to cancel follow-up for a number of contacts in bulk-edit mode though. Maybe that's okay as a workaround?

**Comment by Marko-ilmkreis** [comment]
@MateStrysewske could it be possible to do that by an SQL statement in the database?

**Comment by MateStrysewske** [comment]
@Marko-ilmkreis Yes, that would be possible as well

**Comment by bernardsilenou** [comment]
@MateStrysewske what about implementing  issue #1855. The category called "overdue follow-up" is just what this issue is about. In this way, this issue will not be needed anymore because only active contacts/ cases with ongoing followup will receive reminders messages.

**Comment by MateStrysewske** [comment]
@bernardsilenou I don't think implementing that issue will solve this problem. Setting the follow-up status to overdue will still only make sense on a server where the follow-up calls/visits are actually performed and tracked, which isn't necessarily the case. According to @Marko-ilmkreis, this is exactly the case in Germany right now.

**Comment by kwa20** [comment]
@MateStrysewske @Marko-ilmkreis I wouldn't be too happy with this implementation because by automating this with a server configuration it would mean that it doesn't go through any validation in the system. I would suggest to solve this with an additional bulk action option. However, as @bernardsilenou mentioned, implementing #1855 should be done first because otherwise we'd have to change it all over again after its initial implementation.

**Comment by bernardsilenou** [comment]
Here are some limitations that I can see with this automation of " _If the last date of the follow-up-period is finished, the follow-up must be finished in the system too_"
1. weekend days: contacts may have their end of followup on Friday that is a public holiday. these contacts will automatically be set to followup completed + status = dropped. It is good to have a buffer state (like overdue followup and give users the option to decide how to deal with them)
2. Even if we go with automatically setting the followup to completed, should we also set the contact status to dropped?
  2.1 If yes, then this is contrary to what is specified in #1855, since the suggestion of #1855 is to set the status to inactive.
 2.2 If no, then we end up with contacts with contact status == active and followup status = completed which is contrary to our definition of an active contact.

One nice solution is to implement #1855 + bulk actions (as @kwa20 recommend above).
*  ie users will filter contacts with followup status == overdue from contact directory or by followup until date in the past
*  assign a new contact status other than active (among: lost to followup, dropped, inactive ) to them which will then automatically correct the followup status to  stopped.

This is a figure from #1855, left = contact status, right = followup status, arrow = direction of validation

![image](https://user-images.githubusercontent.com/12996724/102394041-78ad2b80-3fd9-11eb-9e75-53659d556252.png)

**Comment by bernardsilenou** [comment]
@MateStrysewske @Marko-ilmkreis @kwa20 Can we have a short call tomorrow afternoon ?

**Comment by Marko-ilmkreis** [comment]
I'm on sick leave currently, but for a short call could be time tomorrow after 2pm.

I found the the solution from @bernardsilenou nice with the filter and bulk editing method.

**Comment by bernardsilenou** [comment]
@Marko-ilmkreis we refined issue #1855 and when implemented would respond to the needs of this issue.
 Users would need to filter the contact directory for overdue contacts and use the current bulk action to change the followup status to "cancelled fu"  which would then set then set the contact status == closed.
